buildpatches/go-grpc-channelz.patch

```diff
@@ -0,0 +1,94 @@
+diff --git a/servers-page.go b/servers-page.go
+index cf56ba9..18326ec 100644
+--- a/servers-page.go
++++ b/servers-page.go
+@@ -8,11 +8,24 @@ import (
+ 	log "google.golang.org/grpc/grpclog"
+ )
+ 
++type Server struct {
++	Server  *channelzgrpc.Server
++	Sockets *channelzgrpc.GetServerSocketsResponse
++}
++
+ // writeServers writes HTML to w containing RPC servers stats.
+ //
+ // It includes neither a header nor footer, so you can embed this data in other pages.
+ func (h *grpcChannelzHandler) writeServers(w io.Writer) {
+-	if err := serversTemplate.Execute(w, h.getServers()); err != nil {
++	serversRsp := h.getServers()
++	servers := make(map[int64]Server, len(serversRsp.Server))
++	for _, s := range serversRsp.Server {
++		servers[s.Ref.ServerId] = Server{
++			Server:  s,
++			Sockets: h.getServerSockets(s.Ref.ServerId),
++		}
++	}
++	if err := serversTemplate.Execute(w, servers); err != nil {
+ 		log.Errorf("channelz: executing template: %v", err)
+ 	}
+ }
+@@ -32,6 +45,21 @@ func (h *grpcChannelzHandler) getServers() *channelzgrpc.GetServersResponse {
+ 	return servers
+ }
+ 
++func (h *grpcChannelzHandler) getServerSockets(serverID int64) *channelzgrpc.GetServerSocketsResponse {
++	client, err := h.connect()
++	if err != nil {
++		log.Errorf("Error creating channelz client %+v", err)
++		return nil
++	}
++	ctx := context.Background()
++	sockets, err := client.GetServerSockets(ctx, &channelzgrpc.GetServerSocketsRequest{ServerId: serverID})
++	if err != nil {
++		log.Errorf("Error querying GetServerSockets for server %d: %+v", serverID, err)
++		return nil
++	}
++	return sockets
++}
++
+ const serversTemplateHTML = `
+ {{define "server-header"}}
+     <tr classs="header">
+@@ -47,19 +75,19 @@ const serversTemplateHTML = `
+ 
+ {{define "server-body"}}
+     <tr>
+-        <td><a href="{{link "server" .Ref.ServerId}}"><b>{{.Ref.ServerId}}</b> {{.Ref.Name}}</a></td>
+-        <td>{{with .Data.Trace}} {{.CreationTimestamp | timestamp}} {{end}}</td>
+-        <td>{{.Data.CallsStarted}}</td>
+-        <td>{{.Data.CallsSucceeded}}</td>
+-        <td>{{.Data.CallsFailed}}</td>
+-        <td>{{.Data.LastCallStartedTimestamp | timestamp}}</td>
++        <td><a href="{{link "server" .Server.Ref.ServerId}}"><b>{{.Server.Ref.ServerId}}</b> {{.Server.Ref.Name}}</a></td>
++        <td>{{with .Server.Data.Trace}} {{.CreationTimestamp | timestamp}} {{end}}</td>
++        <td>{{.Server.Data.CallsStarted}}</td>
++        <td>{{.Server.Data.CallsSucceeded}}</td>
++        <td>{{.Server.Data.CallsFailed}}</td>
++        <td>{{.Server.Data.LastCallStartedTimestamp | timestamp}}</td>
+ 		<td>
+-			{{range .ListenSocket}}
```

Comment:
sorry small nit, can you leave the listen socket information in the table as well? w/o the listen address displayed it's difficult to differentiate which server corresponds to which port